lib/Router.js

```diff
@@ -250,6 +250,20 @@ Router.prototype.handle = function(request, connection) {
         return request.error(404, "No handlers available for resource " + resourceId);
     }
 
+    if (!!~[Request.METHOD_PATCH, Request.METHOD_POST, Request.METHOD_PUT].indexOf(method)) {
+        var validate = jsen(_.extend({additionalProperties: false}, route.schema));
+        var resource = request.getResource();
+        if (resource) {
+            delete resource.$id;
```

Comment:
Also, it may be better to delete those information from a value copy of resources instead of modifying resource because it may be used down the line.